src/Xamarin.Android.Build.Tasks/Utilities/XAAssemblyResolver.cs

```diff
@@ -0,0 +1,333 @@
+			AssemblyResolver                = this,
+			MetadataImporterProvider        = readerParameters.MetadataImporterProvider,
+			InMemory                        = readerParameters.InMemory,
+			MetadataResolver                = readerParameters.MetadataResolver,
+			ReadingMode                     = readerParameters.ReadingMode,
+			ReadSymbols                     = haveDebugSymbols,
+			ReadWrite                       = readerParameters.ReadWrite,
+			ReflectionImporterProvider      = readerParameters.ReflectionImporterProvider,
+			SymbolReaderProvider            = readerParameters.SymbolReaderProvider,
+			SymbolStream                    = readerParameters.SymbolStream,
+		};
+		try {
+			return LoadFromMemoryMappedFile (filePath, loadReaderParams);
+		} catch (Exception ex) {
+			log.LogWarning ($"Failed to read '{filePath}' with debugging symbols. Retrying to load it without it. Error details are logged below.");
+			log.LogWarning ($"{ex.ToString ()}");
+			loadReaderParams.ReadSymbols = false;
+			return LoadFromMemoryMappedFile (filePath, loadReaderParams);
+		}
+	}
+
+	AssemblyDefinition LoadFromMemoryMappedFile (string file, ReaderParameters options)
+	{
+		// We can't use MemoryMappedFile when ReadWrite is true
+		if (options.ReadWrite) {
+			return AssemblyDefinition.ReadAssembly (file, options);
+		}
+
+		bool origReadSymbols = options.ReadSymbols;
+		MemoryMappedViewStream? viewStream = null;
+		try {
+			// We must disable reading of symbols, even if they were present, because Cecil is unable to find the symbols file when
+			// assembly file name is unknown, and this is precisely the case when reading module from a stream.
+			// Until this issue is resolved, skipping symbol read saves time because reading exception isn't thrown and we don't
+			// retry the load.
+			options.ReadSymbols = false;
+
+			// Create stream because CreateFromFile(string, ...) uses FileShare.None which is too strict
+			using var fileStream = new FileStream (file, FileMode.Open, FileAccess.Read, FileShare.Read, 4096, false);
+			using var mappedFile = MemoryMappedFile.CreateFromFile (fileStream, null, fileStream.Length, MemoryMappedFileAccess.Read, HandleInheritability.None, true);
+			viewStream = mappedFile.CreateViewStream (0, 0, MemoryMappedFileAccess.Read);
+
+			AssemblyDefinition result = ModuleDefinition.ReadModule (viewStream, options).Assembly;
+			viewStreams.Add (viewStream);
+
+			// We transferred the ownership of the viewStream to the collection.
+			viewStream = null;
+
+			return result;
+		} finally {
+			options.ReadSymbols = origReadSymbols;
+			viewStream?.Dispose ();
+		}
+	}
+
+	AssemblyDefinition? SelectAssembly (AndroidTargetArch arch, string assemblyName, CacheEntry? entry, bool loading)
+	{
+		if (entry == null) {
+			// Should "never" happen...
+			throw new ArgumentNullException (nameof (entry));
+		}
+
+		if (arch == AndroidTargetArch.None) {
+			// Disabled for now, generates too much noise.
```

Comment:
It's "concerning" if this is generating too much noise.  I'm not sure what that means, but it's concerning?

Reply:
Because `Cecil` has no idea about architectures and will call us on `Resolve` overloads which don't specify the architecture, so we use `None` and that causes this warning. It won't be an issue when we eventually have one resolver per RID.